### PR TITLE
dockerfile update

### PR DESCRIPTION
**Pull request by leoraba**
Build a docker image ready for production

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Deny All
+*
```
Comment (justincorrigible) [inline .dockerignore:2]: this is not a traditional approach, but I think I like 👍 
clearly implies we're being discerning about what's being published

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# Global variables
+ARG COMMIT=""
+ARG APP_USER=lyric
+ARG WORKDIR=/usr/src/app
+
+######################
+# Configure base image
+######################
+FROM node:18.16.1-alpine AS base
```
Comment (justincorrigible) [inline Dockerfile:9]: It's a new app, let's aim for today's LTS (v20.*)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+COPY --from=pruned ${WORKDIR}/pruned/dist ./dist
+COPY --from=pruned ${WORKDIR}/pruned/node_modules ./node_modules
+
+EXPOSE 3000
+
+ENV COMMIT_SHA=${COMMIT}
+ENV NODE_ENV=production
+
+CMD [ "node", "./dist/src/server.js"]
```
Comment (justincorrigible) [inline Dockerfile:86]:
```suggestion
CMD [ "node", "./dist/src/server.js"]

```

File: .dockerignore [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+!pnpm-lock.yaml
+!pnpm-workspace.yaml
+!tsconfig.base.json
+!tsconfig.json
+
+# Ignore hidden files from sub-directories
+**/.*
+
+# Ignore test folders
+**/test/
```
Comment (justincorrigible) [inline .dockerignore:17]:
```suggestion
**/test/

```

File: Dockerfile (hunk 3 above)
Comment (leoraba) [reply Dockerfile]: added newline

File: .dockerignore (hunk 4 above)
Comment (leoraba) [reply .dockerignore]: added newline

File: Dockerfile (hunk 2 above)
Comment (leoraba) [reply Dockerfile]: updated to version `20.12.2-alpine` currently the latest

**Review by justincorrigible — APPROVED** [review]
Squash all the things :shipit:

**Review by justincorrigible — APPROVED** [review]

**Comment by justincorrigible** [comment]
ready to squash!

**Pull request merged**